### PR TITLE
Enzyme: try bump without tuple changes

### PR DESCRIPTION
**Pull request by wsmoses**
Requires https://github.com/EnzymeAD/Enzyme.jl/pull/1485 to release, which itself requires https://github.com/JuliaRegistries/General/pull/108556

Perhaps means we don't change the other varargs stuff, but we'll have to wait and see.

### CODE REVIEW

File: test/test_enzyme.jl [hunk 1]
```diff
@@ -17,9 +85,186 @@ f(grid) = CenterField(grid)
...
+                set_initial_condition_via_launch!,
+                Duplicated(model_tracer, dmodel_tracer),
+                Active(1.0))
+
+    # Test differentiation of the high-level set interface
+    dmodel = Enzyme.make_zero(model)
+    autodiff(Enzyme.Reverse,
+                set_initial_condition!,
+                Duplicated(model, dmodel),
+                Active(1.0))
```
Comment (glwagner) [inline test/test_enzyme.jl:188]: some indendenting irregularities here and above

File: src/Utils/kernel_launching.jl [hunk 2]
```diff
@@ -113,7 +113,7 @@ end
 Launches `kernel!`, with arguments `args` and keyword arguments `kwargs`,
 over the `dims` of `grid` on the architecture `arch`. kernels run on the default stream
 """
-function launch!(arch, grid, workspec, kernel!, kernel_args...;
+@inline function launch!(arch, grid, workspec, kernel!, kernel_args...;
                  include_right_boundaries = false,
                  reduced_dimensions = (),
                  location = nothing,
```
Comment (glwagner) [inline src/Utils/kernel_launching.jl:119]: fix the indenting here

**Review by glwagner — APPROVED** [review]

**Comment by glwagner** [comment]
seems simple enough, just a few cosmetic comments. Does `shadowp` stand for "`primal_shadow`"?

File: Project.toml [hunk 3]
```diff
@@ -11,6 +11,7 @@ CubedSphere = "7445602f-e544-4518-8976-18f8e8ae6cdb"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 Distances = "b4f34e82-e78d-54a5-968a-f98e89d6e8f7"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
+Enzyme = "7da242da-08ed-463a-9acd-ee780be4f1d9"
```
Comment (vchuravy) [inline Project.toml:14]: Should be in [extras] if it is just a testdep

**Comment by wsmoses** [comment]
@glwagner @jlk9 with a bunch more fixes pushed into Enzyme itself, this now fully passes the Enzyme (cpu) tests so I think we should be good to merge!

**Comment by wsmoses** [comment]
@navidcy @glwagner this should superceded all previous enzyme prs (including compatbwlper), which I've gone ahead and closed just now.

Some non enzyme tests had a nondeterministic ci failure about a directory not being empty, but otherwise all pass (incl enzyme tests).

Once landed we should add a lot more integration test cases like @jlk9 your flux boundary

**Pull request merged**